File: framework/ui/src/io/intino/alexandria/ui/displays/components/DecisionDialog.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.components.selector.Selector;
import io.intino.alexandria.ui.displays.events.SelectionEvent;
import io.intino.alexandria.ui.displays.events.SelectionListener;
import io.intino.alexandria.ui.displays.notifiers.DecisionDialogNotifier;

public class DecisionDialog<DN extends DecisionDialogNotifier, B extends Box> extends AbstractDecisionDialog<DN, B> {
    private Selector selector;
    private SelectionListener selectionListener = null;

    public DecisionDialog(B box) {
        super(box);
    }

    public void bindTo(Selector selector) {
        this.selector = selector;
        selector.onSelect(event -> {
        	if (event.selection().size() == 0) return;
            close();
            notifySelect(event);
        });
    }

    public DecisionDialog onSelect(SelectionListener listener) {
        this.selectionListener = listener;
        return this;
    }

    @Override
    public void open() {
        selector.reset();
        super.open();
    }

    private void notifySelect(SelectionEvent event) {
        if (selectionListener == null) return;
        selectionListener.accept(new SelectionEvent(this, event.selection()));
    }
}

File: framework/ui-elements/src/displays/components/DecisionDialog.js
import React from "react";
import {withStyles} from 'alexandria-ui-elements/src/util/muiStylesCompat';
import {Dialog as MuiDialog} from "@mui/material";
import AbstractDecisionDialog from "../../../gen/displays/components/AbstractDecisionDialog";
import DecisionDialogNotifier from "../../../gen/displays/notifiers/DecisionDialogNotifier";
import DecisionDialogRequester from "../../../gen/displays/requesters/DecisionDialogRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import {withSnackbar} from "alexandria-ui-elements/src/util/notistackCompat";
import BaseDialog from "./BaseDialog";

const styles = theme => ({...BaseDialog.Styles(theme)});

class DecisionDialog extends AbstractDecisionDialog {

	constructor(props) {
		super(props);
		this.notifier = new DecisionDialogNotifier(this);
		this.requester = new DecisionDialogRequester(this);
	};

	render() {
		const handleClose = (event, reason) => {
			if (this.state.modal && reason === "backdropClick") return;
			this.handleClose(event, reason);
		};
		return (
			<MuiDialog fullScreen={this.props.fullscreen} open={this.state.opened}
			           fullWidth={this._widthDefined()} maxWidth={this._widthDefined() ? "xl" : "sm"}
			           onClose={handleClose}
			           slots={this.props.fullscreen ? { transition: this._transition() } : undefined}
			           PaperComponent={!this.props.fullscreen ? this.DraggablePaper : undefined}
                       aria-labelledby={this.props.id + "_draggable"}>
				{this.renderTitle()}
				{this.renderContent(() => <div style={{maxHeight:'300px',overflow:'auto',height:'100%'}}>{this.props.children}</div>)}
			</MuiDialog>
		);
	};

}

export default withStyles(styles, { withTheme: true })(withSnackbar(DecisionDialog));
DisplayFactory.register("DecisionDialog", withStyles(styles, { withTheme: true })(withSnackbar(DecisionDialog)));
